audformat/core/scheme.py

```diff
@@ -65,7 +72,7 @@ def __init__(
             self,
             dtype: typing.Union[typing.Type, define.DataType] = None,
             *,
-            labels: typing.Union[dict, list] = None,
+            labels: typing.Union[dict, list, MiscTable] = None,
```

Comment:
I was expecting we would pass a table ID and not the object itself. Otherwise we cannot know if the table is actually part of the database.

Reply:
We can know, as it contains `self._db`.
I also wanted to start with using the table ID, but for some reason/problem I cannot remember I switched then ;)

But maybe this problem is gone and we can switch back?

Reply:
Ok, but since everywhere else we are using IDs when we refer to objects, I think we should stick to this rule here, too.

Reply:
Access to `self._db` we only have after we assigned the scheme to the database. But to assign it, we first need to create it.

Reply:
Then we should add `Scheme.db` as with `Table.db`.

Reply:
We have access to `db` inside `Scheme`, but only after we have assigned the scheme to a database.
The only way around would be to first create an empty scheme, assign it to a database and then update the labels with a misc table. Or we could add a `db` argument to `Scheme` which is not very nice.

Reply:
> We have access to db inside Scheme, but only after we have assigned the scheme to a database.

I see, we need to access `MiscTable` already in the constructor to derive `dtype`....

Reply:
We could add a new `dtype` for `MiscTable`.

Reply:
E.g. `dtype='table'`

Reply:
We would also need to remove all checks that the given table is valid and that the table ID exists at all in the database.
In addition, we could not output the YAML of the misc table until the scheme is assigned to a database.

Reply:
> We would also need to remove all checks that the given table is valid and that the table ID exists at all in the database.

We can those checks here:  https://github.com/audeering/audformat/blob/ee93219b24c3b428d5ab0ded5f56ab72bb866101/audformat/core/database.py#L1010

> In addition, we could not output the YAML of the misc table until the scheme is assigned to a database.

Which is fine I think, since this is anyway duplicated information.

Reply:
Makes indeed sense.

How would you then handle checking for dtypes, and handling of defined dtypes, e.g.
```python
define.DataType.assert_has_attribute_value(dtype)
```
Should we then check for the right dtypes only when the scheme is assigned to the database, or should we simply don't care and go with dtype `table`?

I guess for the YAML representation you would then go simply with
```python
>>> Scheme(labels='table_id')
dtype: table
labels: table_id
```

Reply:
an alternative to `dtype='table'` would be to force a user to provide a `dtype` when assigning a table ID and then check for the correct dtypes in the index when the scheme is assigned to the database.

Reply:
> an alternative to dtype='table' would be to force a user to provide a dtype when assigning a table ID and then check for the correct dtypes in the index when the scheme is assigned to the database.

Yes, that also sounds like a good solution.

Reply:
I switched to using a table ID to assign a misc table as scheme and require now indeed that `dtype` is provided in this case.